### ISSUE TITLE
sshd detects normal logins and bans after threshold

### ISSUE THREAD

**Issue by gbeekmans** [fail2ban/fail2ban#2632]
The filter.d/sshd.conf file currently contains the following in cmnfailre:
```
^<F-MLFFORGET><F-MLFGAINED>Accepted \w+</F-MLFGAINED></F-MLFFORGET> for <F-USER>\S+</F-USER> from <HOST>(?:\s|$)
```
This causes a detection upon normal login (tested with public keys). After a few successful logins in a row, the IP is banned. Removing that regex stops this from happening.

Is this intentional or is there an actual issue that needs addressing?

**Comment by sebres** [comment]
People! In our issue template we are asking for fail2ban version not for nothing.

If you use newest sshd-filter with fail2ban version lower than 0.10.5, it is wrong (one should use filter previously #2279).
In newer version the tag `<F-MLFGAINED>...</F-MLFGAINED>` (if not empty) causes positive match, which means a successful login was detected and it is made exactly an opposite to your assumption 
- it forgets all failures previously made within same sessions (if user name is the same).
The user check is very important, because it avoids brute force of some legitimate users (e. g. user `tester`) on other accounts (e. g. of user `root`) so on gained access it would forget only the failures made by this IP for user `tester`.

This makes possible to lower the bruteforce attempts without false positives using multiple public keys.
See https://github.com/fail2ban/fail2ban/issues/2188#issuecomment-407425179 for details.

Anyway I don't think it is the case (at least in fail2ban >= 0.10.5).
So this session doesn't cause failures in fail2ban:
```
Nov 28 09:16:03 srv sshd[32307]: Failed publickey for git from 192.0.2.1 port 57904 ssh2: ECDSA 0e:ff:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx
Nov 28 09:16:03 srv sshd[32307]: Failed publickey for git from 192.0.2.1 port 57904 ssh2: RSA 04:bc:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx
Nov 28 09:16:03 srv sshd[32307]: Postponed publickey for git from 192.0.2.1 port 57904 ssh2 [preauth]
Nov 28 09:16:03 srv sshd[32307]: Accepted publickey for git from 192.0.2.1 port 57904 ssh2: DSA 36:48:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx
```
But this must find that:
```
Nov 28 09:16:05 srv sshd[32310]: Failed publickey for git from 192.0.2.2 port 57910 ssh2: ECDSA 1e:fe:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx
Nov 28 09:16:05 srv sshd[32310]: Failed publickey for git from 192.0.2.2 port 57910 ssh2: RSA 14:ba:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx
Nov 28 09:16:05 srv sshd[32310]: Disconnecting: Too many authentication failures for git [preauth]
Nov 28 09:16:05 srv sshd[32310]: Connection closed by 192.0.2.2 [preauth]
```
Fail2ban writes all the matched attempts in database, so if you use correct version and you mean it does affect you - please provide the log excerpt for the IP causing that as well as result of following command (replace `192.0.2.1` with IP causing this issue):
```
?sudo? sqlite3 -header -column 'file:/var/lib/fail2ban/fail2ban.sqlite3?mode=ro' "select * from bans where jail = 'sshd' and ip = '192.0.2.1'"
```

Without this info we have no chance to find what exactly happens on your side.

**Comment by gbeekmans** [comment]
This issue can be closed as invalid. I was under the impression I successfully upgraded to version 0.10.5 but evidently the host is still on 0.10.4 (or it was downgraded after a vmware snapshot rollback). Re-tested with 0.10.5 and it works as designed. Sorry to waste your time. :)
